src/source.rs

```diff
@@ -79,11 +80,17 @@ impl<S: AsRef<str>> From<S> for Source {
                 })
                 .collect(),
             len: offset,
+            line_offset: 0,
         }
     }
 }
 
 impl Source {
+    /// Add an offset to the line number displayed in the message.
+    pub fn with_line_offset<S: AsRef<str>>(source: S, line_offset: usize) -> Source {
+        Source { line_offset, ..Source::from(source) }
```

Comment:
`with_x` might make more sense as a builder pattern (i.e: takes `self`, returns `Self`). This would be strictly more powerful.